en/02_Developer_Guides/02_Controllers/02_Routing.md

```diff
@@ -303,6 +303,97 @@ class BreadAPIController extends Controller
 In Silverstripe CMS versions prior to 4.6, an empty key (`''`) must be used in place of the `'/'` key. When specifying an HTTP method, the empty string must be separated from the method (e.g. `'GET '`). The empty key and slash key are also equivalent in Director rules.
 [/alert]
 
+## Nested request handlers
+
+Nested [`RequestHandler`](api:SilverStripe\Control\RequestHandler) routing is used extensively in the CMS and is used to create URL endpoints without YAML configuration. Nesting is done by returning a `RequestHandler` from an action method on another `RequestHandler`, usually a `Controller`.
+
+`RequestHandler` is the base class for other classes that can handle HTTP requests such as `Controller`, [`FormRequestHandler`](api:SilverStripe\Forms\FormRequestHandler) (used by [`Form`](api:SilverStripe\Forms\Form)) and [`FormField`](api:SilverStripe\Forms\FormField).
+
+### How it works
+
+[`Director::handleRequest()`](api:SilverStripe\Control\Director::handleRequest()) begins the URL parsing process by parsing the start of the URL and workng out which request handler subclass to use by looking in routes set in YAML config under `Director.rules`.
+
+When a request handler matching the first portion of the URL is found, the `handleRequest()` method on the matched request handler subclass is called. This passes control to the matched request handler and the URL is effectively truncated from the left, removing the portion that lead to ths point.
+
+From there regular request handling occurs and the URL will be checked to see if it matches `$allowed_actions` on the `RequestHandler`, possibly routed via `$url_handlers`. If an `$allowed_action` (i.e. method on the `RequestHandler`) is matched and that method returns a request handler, then control will be passed to this nested request handler and the URL is again effectively truncated from the left.
+
+### Example of anested request handler being returned in an action method
+
+Using the code below, navigating to the URL `/one/two/hello` will return a response with a body of "hello"
+
+```yml
+// app/_config/routes.yml
+
+SilverStripe\Control\Director:
+  rules:
+    'one': 'RequestHandlerOne'
+```
+
+```php
+// app/src/RequestHandlerOne.php
+
+use SilverStripe\Control\Controller;
+use SilverStripe\Control\HTTPRequest;
+
+class RequestHandlerOne extends Controller
+{
+    private static $allowed_actions = [
+        'two',
+    ];
+
+    public function two(HTTPRequest $request)
+    {
+        return RequestHandlerTwo::create();
+    }
+}
+```
+
+```php
+// app/src/RequestHandlerTwo.php
+
+use SilverStripe\Control\Controller;
+use SilverStripe\Control\HTTPRequest;
+use SilverStripe\Control\HTTPResponse;
+
+class RequestHandlerTwo extends Controller
+{
```

Comment:
I've added this though I can't help thinking that it's kind of implied anyway since they're just examples.  Other examples in the file don't have these, I don't think they're necessary or add anything